src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1588,30 +1608,31 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 				//
 				case IntrinsicId.MethodInfo_MakeGenericMethod: {
 						reflectionContext.AnalyzingPattern ();
+						if (callingMethodDefinition.Name == "TestWithRequirementsButNoTypeArguments")
+							Debug.WriteLine ("");
 
+						bool hasTypeArguments = (methodParams[1] as ArrayValue)?.Size.AsConstInt () != 0;
 						foreach (var methodValue in methodParams[0].UniqueValues ()) {
 							if (methodValue is SystemReflectionMethodBaseValue methodBaseValue) {
-								foreach (var genericParameter in methodBaseValue.MethodRepresented.GenericParameters) {
-									if (_context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (genericParameter) != DynamicallyAccessedMemberTypes.None ||
-										genericParameter.HasDefaultConstructorConstraint) {
-										// There is a generic parameter which has some requirements on input types.
-										// For now we don't support tracking actual array elements, so we can't validate that the requirements are fulfilled.
-										reflectionContext.RecordUnrecognizedPattern (
-											2060, string.Format (Resources.Strings.IL2060,
-												DiagnosticUtilities.GetMethodSignatureDisplayName (calledMethod)));
-									}
-								}
-
-								// We haven't found any generic parameters with annotations, so there's nothing to validate
+								ValidateGenericMethodInstantiation (ref reflectionContext, methodBaseValue.MethodRepresented, methodParams[1], calledMethod);
 								reflectionContext.RecordHandledPattern ();
 							} else if (methodValue == NullValue.Instance) {
 								reflectionContext.RecordHandledPattern ();
 							} else {
-								// There is a generic parameter which has some requirements on input types.
-								// For now we don't support tracking actual array elements, so we can't validate that the requirements are fulfilled.
-								reflectionContext.RecordUnrecognizedPattern (
-									2060, string.Format (Resources.Strings.IL2060,
-										DiagnosticUtilities.GetMethodSignatureDisplayName (calledMethod)));
+								if (hasTypeArguments) {
```

Comment:
Is it worth it to add big comment blocks and extra code just to avoid generating a warning when someone calls MakeGenericMethod with an empty array?

Reply:
Probably not - originally I did it because I've done it for `Expression.Call` (where it's much more common). I'll remove this code.